apis/bigquery/v1beta1/table_reference.go

```diff
@@ -71,10 +72,30 @@ func (r *TableRef) NormalizedExternal(ctx context.Context, reader client.Reader,
 		return "", fmt.Errorf("reading referenced %s %s: %w", BigQueryTableGVK, key, err)
 	}
 	// Get external from status.externalRef. This is the most trustworthy place.
-	actualExternalRef, _, err := unstructured.NestedString(u.Object, "status", "externalRef")
+	actualExternalRef, found, err := unstructured.NestedString(u.Object, "status", "externalRef")
 	if err != nil {
 		return "", fmt.Errorf("reading status.externalRef: %w", err)
 	}
+	// For non-direct resources, there's no status.externalRef
+	if !found {
```

Comment:
This feels like a common pattern we need to handle when migrating BETA resources. We should investigate if it is possible to build a general helper function to ensure best practice. Related: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/65b84c3dba0bfe378ec521f70df07977cfc9efc9/apis/bigquery/v1beta1/dataset_reference.go#L78-L100

Reply:
/cc @acpana